packages/jest-haste-map/src/crawlers/__tests__/watchman.test.js

```diff
@@ -13,74 +13,82 @@ const path = require('path');
 jest.mock('fb-watchman', () => {
   const normalizePathSep = require('../../lib/normalize_path_sep').default;
   const Client = jest.fn();
-  Client.prototype.command = jest.fn((args, callback) => {
-    if (args[0] === 'watch-project') {
-      setImmediate(() => callback(null, {watch: args[1].replace(/\\/g, '/')}));
-    } else if (args[0] === 'query') {
-      setImmediate(() =>
-        callback(null, mockResponse[normalizePathSep(args[1])]),
-      );
-    }
-  });
+  Client.prototype.command = jest.fn((args, callback) =>
+    setImmediate(() => {
+      const response = mockResponse[args[0]][normalizePathSep(args[1])];
+      callback(null, response.next ? response.next().value : response);
+    }),
+  );
   Client.prototype.on = jest.fn();
   Client.prototype.end = jest.fn();
   return {Client};
 });
 
+const forcePOSIXPaths = path => path.replace(/\\/g, '/');
 const pearMatcher = path => /pear/.test(path);
 
+let watchman;
 let watchmanCrawl;
 let mockResponse;
 let mockFiles;
 
-const FRUITS = path.sep + 'fruits';
-const VEGETABLES = path.sep + 'vegetables';
+const ROOT_MOCK = path.sep === '/' ? '/root-mock' : 'M:\\root-mock';
+const FRUITS = `${ROOT_MOCK}${path.sep}fruits`;
+const VEGETABLES = `${ROOT_MOCK}${path.sep}vegetables`;
 const ROOTS = [FRUITS, VEGETABLES];
 const BANANA = path.join(FRUITS, 'banana.js');
 const STRAWBERRY = path.join(FRUITS, 'strawberry.js');
 const KIWI = path.join(FRUITS, 'kiwi.js');
 const TOMATO = path.join(FRUITS, 'tomato.js');
 const MELON = path.join(VEGETABLES, 'melon.json');
+const WATCH_PROJECT_MOCK = {
+  [FRUITS]: {
+    relative_path: 'fruits',
+    watch: forcePOSIXPaths(ROOT_MOCK),
```

Comment:
Watchman always returns forward slashes and `path.join()` uses `\` as path separator on Windows. To correctly mimic Watchman responses, we need this. Am I missing something?

Reply:
no, I just missed that information in this context. All good :-)

Reply:
`path.posix.join`?

Reply:
I don't think that replaces existing backslashes.
```
> path.posix.join('M:\\lol')
'M:\\lol'
```

Reply:
Ah, good point!

Could use `slash`, not sure if better, though.

```
> const slash = require('slash');
undefined
'M:/lol'
> slash('M:\\lol')
'M:/lol'
```

Reply:
Not sure if it is worth the effort or the extra `devDependency` only for tests and to support UNC paths but if you think this would be an improvement, I can submit another PR?

Reply:
Nah, it's fine (we have slash already, fwiw)

Reply:
>we have slash already, fwiw

Didn't see it in `jest-haste-map`, that's why :D